packages/cli/src/cmds/validator/keys.ts

```diff
@@ -60,12 +60,24 @@ export async function getSecretKeys(
       lockFile.lockSync(lockFilePath);
     }
 
-    const secretKeys = await Promise.all(
-      keystorePaths.map(async (keystorePath) =>
-        SecretKey.fromBytes(await Keystore.parse(fs.readFileSync(keystorePath, "utf8")).decrypt(passphrase))
-      )
-    );
+    const secretKeysPromise = keystorePaths.map(async (keystorePath) => {
```

Comment:
A better solution would be to filter the `keystorePaths` by known name formats. Specifically excluding files that look like the deposit data json: `deposit_data-*.json`

Reply:
It also must apply to any place that keystores are read from disk, specially here https://github.com/ChainSafe/lodestar/blob/c581c18a0fd2f9db867c4f9852651bea64c51e4a/packages/cli/src/cmds/validator/keys.ts#L52

Reply:
> A better solution would be to filter the `keystorePaths` by known name formats. Specifically excluding files that look like the deposit data json: `deposit_data*.json`

I thought about this, but was not sure if the pattern of the filename should be the determinant or the content of the file itself? Like would the filename pattern always remain this way? If that is a yes then I can update to filter using the file name

Reply:
Also there could be situations where the file has the correct file name but fails to parse for whatever reason. The application should still not choke on startup but should emit a warning. So I’m still leaning on using the content as being a more robust approach. Let me know what you think.

Reply:
> The application should still not choke on startup but should emit a warning

The application must error! This is a validator client, if it can not decrypt the keys it can not do anything

Reply:
We can use the pattern `deposit_data*.json` safely because the only key generator tool recommended by all clients uses this very specific convention

https://github.com/ethereum/eth2.0-deposit-cli/blob/256ea2148cd088d6a8672bd0a3f5b12617159960/eth2deposit/credentials.py#L199

Reply:
> We can use the pattern `deposit_data*.json` safely because the only key generator tool recommended by all clients uses this very specific convention

Gotcha!